doc/high-level.md

```diff
@@ -0,0 +1,83 @@
+| `(rand '(3 4))`                                 | `rand(3,4)`        | `random.rand(3,4)`                | Create a random 3x4 array.                                                           |
+
+### Basic Operations
+
+| MAGICL     | MATLAB   | NumPy            | Description                 |
+|------------|----------|------------------|-----------------------------|
+| `(@ a b)`  | `a * b`  | `a @ b`          | Matrix multiplication       |
+| `(.+ a b)` | `a + b`  | `a + b`          | Element-wise add            |
+| `(.- a b)` | `a - b`  | `a - b`          | Element-wise subtract       |
+| `(.* a b)` | `a .* b` | `a * b`          | Element-wise multiply       |
+| `(./ a b)` | `a./b`   | `a/b`            | Element-wise divide         |
+| `(.^ a b)` | `a.^3`   | `a**3`           | Element-wise exponentiation |
+
+### Linear Algebra
+
+| MAGICL                                                  | MATLAB            | NumPy                                                          | Description                                |
+|---------------------------------------------------------|-------------------|----------------------------------------------------------------|--------------------------------------------|
+| `(det a)`                                               | `det(a)`          | `linalg.det(a)`                                                | Determinant of matrix                      |
+| `(trace a)`                                             | `trace(a)`        | `trace(a)`                                                     | Trace (sum of diagonal elements) of matrix |
+| `(upper-triangular a)`                                  | `triu(a)`         | `triu(a)`                                                      | Upper triangular part of matrix            |
+| `(lower-triangular a)`                                  | `tril(a)`         | `tril(a)`                                                      | Lower triangular part of matrix            |
+| `(transpose a)`                                         | `a.'`             | `a.transpose()` or `a.T`                                       | Transpose of matrix                        |
+| `(conjugate-transpose a)` or `(dagger a)`               | `a'`              | `a.conj().transpose()` or `a.conj().T`                         | Conjugate transpose of matrix              |
+| `(inv a)`                                               | `inv(a)`          | `linalg.inv(a)`                                                | Inverse of matrix                          |
+| `(svd a)` (Returns `(VALUES U SIGMA Vt)`)               | `[U,S,V]=svd(a)`  | `U, S, Vh = linalg.svd(a), V = Vh.T`                           | Singular value decomposition of matrix     |
+| `(eig a)` (Returns `(VALUES EIGENVALUES EIGENVECTORS)`) | `[V,D]=eig(a)`    | `D,V = linalg.eig(a)`                                          | Eigenvalues and eigenvectors of matrix     |
```

Comment:
This seems to only track the real parts of the eigenvalues and eigenvalues. For example, `(from-list '(0d0 -1d0 1d0 0d0) '(2 2))` is a 90 degree counterclockwise rotation. It has no real eigenvalues, but it is orthogonal and has pure imaginary eigenvalues. 
```
MAGICL> (eig (from-list '(0d0 -1d0 1d0 0d0) '(2 2)))
(0.0d0 0.0d0)
#<MATRIX/DOUBLE-FLOAT (2x2):
   0.707     0.000
   0.000    -0.707>
```
Compare with numpy:
```
In [7]: np.linalg.eig(np.array([[0, 1.0], [-1.0, 0.0]]))
Out[7]:
(array([0.+1.j, 0.-1.j]),
 array([[0.70710678+0.j        , 0.70710678-0.j        ],
        [0.        +0.70710678j, 0.        -0.70710678j]]))
```

Reply:
Seems to work when the type is complex.
```
MAGICL> (eig (from-list '(0d0 -1d0 1d0 0d0) '(2 2) :type '(complex double-float)))
(#C(0.0d0 0.9999999999999997d0) #C(2.7755575615628914d-17 -1.0d0))
#<MATRIX/COMPLEX-DOUBLE-FLOAT (2x2):
   0.707 + 0.000j     0.707 + 0.000j
  -0.000 - 0.707j    -0.000 + 0.707j>
```

I think this brings up the question of if `eig` on a real tensor should be allowed to return complex results. I am fine with it doing that and it shouldn't be too hard to implement.

Reply:
I think `eig` should be allowed to return complex results. FWIW it's not a bad idea to have a variant for real symmetric (or complex hermitian) matrices -- this is an approach taken by numpy (e.g. `linalg.eigh`) as well as underlying lapack routines.

Reply:
(Also, for the purposes of getting this PR over the finish line, I don't view adding any more top-level routines as a requirement, but it's worth keeping in mind for the future.)

Reply:
FWIW there is a `hermitian-eig` defined. It should be added to the documentation